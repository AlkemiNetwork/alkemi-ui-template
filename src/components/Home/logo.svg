<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 206 51.33"><defs><style>.cls-1{fill:#231f20;}.cls-2{fill:#578ba6;}.cls-3{fill:#fff;}</style></defs><title>Alkemi-logo</title><path class="cls-1" d="M67.78,18.19l8.57,18.46H73.06L71.74,33.8H62.18l-1.34,2.85H57.55l8.68-18.46ZM63.61,30.78h6.73L67,23.57Z"/><path class="cls-1" d="M97.7,33.62v3H87.45V18.19h3.05V33.62Z"/><path class="cls-1" d="M121.75,36.65l-6.16-7.19-2.66,3v4.22h-3V18.22h3v9.65l8.72-9.68h4.06l-8.06,9,8,9.49Z"/><path class="cls-1" d="M149.74,21.25H141v4.66h5.8v3H141v4.69h8.7v3H138V18.19h11.73Z"/><path class="cls-1" d="M178.19,18.19h2.36V36.65h-3V24.54L171,35.19l-6.62-10.71V36.65h-3V18.19h2.38L171,29.88Z"/><rect class="cls-1" x="191.72" y="17.97" width="3.34" height="18.3"/><circle class="cls-2" cx="27.05" cy="27.49" r="21.5"/><path class="cls-3" d="M12.94,37.06l-.42.05Z"/><path class="cls-3" d="M13.49,37l.27,0h0Z"/><path class="cls-3" d="M12,37.18l-.34,0Z"/><path class="cls-3" d="M22.94,31.55h0l.23-.25Z"/><path class="cls-3" d="M23.54,30.87l.35-.43Z"/><path class="cls-3" d="M30.22,31.32l-.27,0Z"/><path class="cls-3" d="M40.71,37l-.21-.24h0Z"/><path class="cls-3" d="M41.62,38.2c.16.23.31.47.46.71C41.93,38.67,41.78,38.43,41.62,38.2Z"/><path class="cls-3" d="M31.27,31.52,31,31.47Z"/><path class="cls-3" d="M41,37.41l.34.44Z"/><path class="cls-3" d="M13.76,36.91l.16,0-.16,0Z"/><path class="cls-3" d="M22.81,31.7l.14-.15h0Z"/><path class="cls-3" d="M19.52,34.48l-.39.26Z"/><path class="cls-3" d="M22.42,32.11l-.38.37Z"/><path class="cls-3" d="M18.64,35l-.41.23Z"/><path class="cls-3" d="M17.73,35.52l-.42.2Z"/><path class="cls-3" d="M21.63,32.86l-.39.33Z"/><path class="cls-3" d="M20.28,34l-.27.2Z"/><path class="cls-3" d="M16.8,36l-.43.18Z"/><path class="cls-3" d="M14.89,36.63l-.43.12Z"/><path class="cls-3" d="M15.85,36.32l-.43.15Z"/><path class="cls-3" d="M32.08,31.71l.18.05Z"/><path class="cls-3" d="M39,35.28l.33.29Z"/><path class="cls-3" d="M38.17,34.66l.33.25Z"/><path class="cls-3" d="M39.69,35.95l.33.33Z"/><path class="cls-3" d="M37.36,34.08l.31.21Z"/><path class="cls-3" d="M35.79,33.15l.12.06Z"/><path class="cls-3" d="M36.54,33.57l.27.16Z"/><path class="cls-3" d="M40.49,36.79h0l-.12-.13Z"/><path class="cls-3" d="M42.08,38.91c-.15-.24-.3-.48-.46-.71l-.25-.34L41,37.41,40.71,37l-.21-.25-.11-.13L40,36.28l-.33-.33-.41-.38L39,35.28l-.45-.37-.33-.25-.5-.36-.31-.21-.55-.36-.27-.16-.64-.36-.12-.06a18,18,0,0,0-3.53-1.39l-.18-.05-.81-.19h0L31,31.47l-.81-.14-.27,0-.82-.1a18.32,18.32,0,0,0-4.78.11A17.24,17.24,0,0,1,12.93,38c-.25.35-.49.7-.71,1.07l4.27,2.27a12.21,12.21,0,0,1,10.7-5.83,12.21,12.21,0,0,1,10.7,5.83L42.15,39Z"/><polygon class="cls-3" points="11 37.23 11.01 37.23 11 37.23 11 37.23"/><path class="cls-3" d="M26.54,12.45c-.17-.39-.36-.78-.56-1.15l-4.1,2.56A12.21,12.21,0,0,1,21.57,26a12.21,12.21,0,0,1-10.4,6.35L11,37.23l.62,0,.34,0,.57-.07.42-.05.55-.1.27-.05.16,0,.54-.13.43-.12.53-.16.43-.15.52-.2L16.8,36l.51-.23.42-.2.5-.26.41-.23.49-.29.39-.26.49-.32.27-.2q.5-.37,1-.77l.39-.33.41-.39.38-.37.39-.41.13-.15.23-.25.36-.43.35-.43.34-.45a18.49,18.49,0,0,0,2.37-4.31A17.24,17.24,0,0,1,26.54,12.45Z"/><path class="cls-3" d="M42.93,32.25a12.21,12.21,0,0,1-10.4-6.35,12.21,12.21,0,0,1-.3-12.18l-4.1-2.56A14.92,14.92,0,0,0,27,13.69a17.29,17.29,0,0,0,0,10.64,18.91,18.91,0,0,0,.83,2.14q.37.82.83,1.61a19.23,19.23,0,0,0,1.61,2.35A17.24,17.24,0,0,1,41.82,37c.43,0,.85.08,1.28.09Z"/></svg>
