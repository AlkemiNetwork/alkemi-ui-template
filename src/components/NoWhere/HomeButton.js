import React from "react";
import PropTypes from "prop-types";

import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
    button: {
        margin: theme.spacing.unit
    }
});

const NoWhereHomeButton = ({ classes, onClick }) => (
    <Button
        variant="raised"
        color="default"
        className={classes.button}
        onClick={onClick}
    >
        Bring me back home!
    </Button>
);

NoWhereHomeButton.propTypes = {
    classes: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
};

export default withStyles(styles)(NoWhereHomeButton);
