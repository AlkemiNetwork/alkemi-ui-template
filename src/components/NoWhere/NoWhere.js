import React from "react";
import PropTypes from "prop-types";

import ReportIcon from "@material-ui/icons/Report";
import { withStyles } from "@material-ui/core/styles";
import { grey } from "@material-ui/core/colors";

import NoWhereHomeButton from "../../containers/NoWhere/HomeButton";

const styles = theme => ({
    nowhere: {
        textAlign: "center"
    },
    nowhereHeader: {
        padding: theme.spacing.unit * 3
    },
    nowhereTitle: {
        fontSize: "1.5em",
        color: grey[500]
    },
    nowhereIcon: {
        height: 80,
        width: 80,
        color: grey[500]
    }
});

const NoWhere = ({ classes }) => (
    <div className={classes.nowhere}>
        <div className={classes.nowhereHeader}>
            <ReportIcon className={classes.nowhereIcon} />
            <h1 className={classes.nowhereTitle}>
                Oups... It seems that you are lost!
            </h1>
        </div>
        <NoWhereHomeButton />
    </div>
);

NoWhere.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(NoWhere);
