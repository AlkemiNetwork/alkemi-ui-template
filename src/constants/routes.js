// router path
export const HOME = "/home";
