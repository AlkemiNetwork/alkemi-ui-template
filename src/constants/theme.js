import { createMuiTheme } from "@material-ui/core/styles";
import indigo from "@material-ui/core//colors/indigo";
import green from "@material-ui/core/colors/green";
import red from "@material-ui/core//colors/red";

const theme = createMuiTheme({
    palette: {
        primary: indigo,
        secondary: {
            ...green,
            A400: "#00e677"
        },
        error: red
    }
});

export default theme;
