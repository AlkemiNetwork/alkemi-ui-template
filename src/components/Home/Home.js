import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";

import logo from "./logo.svg";

const styles = theme => ({
    home: {
        textAlign: "center"
    },
    homeLogo: {
        height: "80px"
    },
    homeHeader: {
        height: "150px",
        padding: "20px"
    },
    homeTitle: {
        fontSize: "1.5em"
    },
    homeIntro: {
        fontSize: "large"
    }
});

const Home = ({ classes }) => (
    <div className={classes.home}>
        <div className={classes.homeHeader}>
            <img src={logo} className={classes.homeLogo} alt="logo" />
            <h1 className={classes.homeTitle}>
                Welcome to ConsenSys React Boilerplate
            </h1>
        </div>
        <p className={classes.homeIntro}>
            To get started, edit <code>src/App.js</code> and save to reload.
        </p>
    </div>
);

Home.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Home);
