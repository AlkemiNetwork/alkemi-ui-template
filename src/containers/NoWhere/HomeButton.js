import { connect } from "react-redux";
import { push } from "connected-react-router";

import NoWhereHomeButton from "../../components/NoWhere/HomeButton";
import { HOME } from "../../constants/routes";

const mapDispatchToProps = dispatch => ({
    onClick: () => dispatch(push(HOME))
});

export default connect(
    undefined,
    mapDispatchToProps
)(NoWhereHomeButton);
