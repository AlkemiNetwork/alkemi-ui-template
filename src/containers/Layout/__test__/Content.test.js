import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Route, Switch } from "react-router";

import LayoutContent from "../Content";
import Home from "../../../components/Home/Home";
import NoWhere from "../../../components/NoWhere/NoWhere";
import { HOME } from "../../../constants/routes";

describe("<LayoutContent />", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LayoutContent />);
    });

    it("renders 1 <Switch /> components", () => {
        expect(wrapper.find(Switch).length).toEqual(1);
    });

    it("renders 2 <Route /> components", () => {
        expect(wrapper.find(Route).length).toEqual(2);
    });

    it("route 1 renders a <Home /> components", () => {
        expect(
            wrapper
                .find(Route)
                .at(0)
                .props().path
        ).toEqual(HOME);
        expect(
            wrapper
                .find(Route)
                .at(0)
                .props().component
        ).toEqual(Home);
    });

    it("route 2 renders a <NoWhere /> components", () => {
        expect(
            wrapper
                .find(Route)
                .at(1)
                .props().component
        ).toEqual(NoWhere);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
