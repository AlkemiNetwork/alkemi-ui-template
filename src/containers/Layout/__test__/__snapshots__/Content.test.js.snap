// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutContent /> matches snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/home"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
